src/v/cloud_storage/async_manifest_view.cc

```diff
@@ -743,6 +743,12 @@ async_manifest_view::get_term_last_offset(model::term_id term) noexcept {
     } else if (stmm.get_archive_start_offset() != model::offset{}) {
         const auto spill_index = get_spillover_upper_bound_by_term(term);
         if (!spill_index.has_value()) {
+            // Fall back to searching the main manifest for term.
+            for (const auto& p : stmm) {
```

Comment:
I don't think we need to iterate here. On line 723 we're checking `stmm.begin()->segment_term <= term`. If this is not the case we're going into this code path. So the invariant here is that `stmm.begin()->segment_term > term`.

Also, the cloud storage log may have gaps so we shouldn't expect to see `term() + 1` here. We may see `term() + 2` or `+10` or whatever. This breaks some api guarantees but we can't really do anything here if TS was disable for long enough and then re-enabled.

So basically, I think it's enough to just return a base offset of the last segment - 1 here without going through the manifest. It's also guaranteed that the main manifest (or STM manifest) is not empty because you have spillover manifests and this code is not async.

Reply:
shouldn't we find the spillover manifest [3] in this case?